src/compiler/transformers/es2015.ts

```diff
@@ -3993,18 +3993,21 @@ namespace ts {
          *
          * @param node The ES6 template literal.
          */
-        function getRawLiteral(node: LiteralLikeNode) {
+        function getRawLiteral(node: TemplateLiteralLikeNode) {
             // Find original source text, since we need to emit the raw strings of the tagged template.
             // The raw strings contain the (escaped) strings of what the user wrote.
             // Examples: `\n` is converted to "\\n", a template string with a newline to "\n".
-            let text = getSourceTextOfNodeFromSourceFile(currentSourceFile, node);
-
-            // text contains the original source, it will also contain quotes ("`"), dolar signs and braces ("${" and "}"),
-            // thus we need to remove those characters.
-            // First template piece starts with "`", others with "}"
-            // Last template piece ends with "`", others with "${"
-            const isLast = node.kind === SyntaxKind.NoSubstitutionTemplateLiteral || node.kind === SyntaxKind.TemplateTail;
-            text = text.substring(1, text.length - (isLast ? 1 : 2));
+            let text = node.rawText;
+            if (text === undefined) {
+                text = node.text;
+
+                // text contains the original source, it will also contain quotes ("`"), dolar signs and braces ("${" and "}"),

```

Comment:
Wait, what? `node.text` is the cooked string value, I thought - yes, it can contain quotes and stuff, but it shouldn't need to be `substring`'d?

Reply:
My bad, that should be `text = getSourceTextOfNodeFromSourceFile(currentSourceFile, node);`